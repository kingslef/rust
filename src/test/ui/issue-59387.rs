trait Object<U> {
    type Output;
}

impl<T: ?Sized, U> Object<U> for T {
    // ^-- Here is the blanket impl; relies on `?Sized`.
    type Output = U;
}

fn foo<T: ?Sized, U>(x: <T as Object<U>>::Output) -> U {
    x
}

fn transmute<T, U>(x: T) -> U {
    foo::<dyn Object<U, Output = T>, U>(x)
    //~^ ERROR the trait `Object` cannot be made into an object
}

fn main() {}
