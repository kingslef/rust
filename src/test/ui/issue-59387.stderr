error[E0038]: the trait `Object` cannot be made into an object
  --> $DIR/issue-59387.rs:15:11
   |
LL | / impl<T: ?Sized, U> Object<U> for T {
LL | |     // ^-- Here is the blanket impl; relies on `?Sized`.
LL | |     type Output = U;
LL | | }
   | |_- traits with non-method items cannot have an impl with an unsized `Self` type
...
LL |       foo::<dyn Object<U, Output = T>, U>(x)
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Object` cannot be made into an object

error: aborting due to previous error

For more information about this error, try `rustc --explain E0038`.
